src/pybind/mgr/ansible/module.py

```diff
@@ -673,6 +694,120 @@ def remove_host(self, host):
 
         return ARSChangeOperation(self.ar_client, self.log, operations)
 
+# JMO --------------------------------------------------------------------------
+    def add_stateless_service(self, service_type, spec):
+        """ Add a stateless service in the cluster
+
+        : service_type: Kind of service (nfs, rgw, mds)
+        : spec        : an Orchestrator.StatelessServiceSpec object
+
+        : returns     : Completion object
+        """
+
+        # Check service_type is supported
+        if service_type not in ["rgw"]:
+            raise orchestrator.OrchestratorError(
+                "{} service not supported".format(service_type))
+
+        # Add the hosts to the inventory in the right group
+        group = "{}s".format(service_type)
+        hosts = spec.service_spec.hosts
+        if not hosts:
+            raise orchestrator.OrchestratorError("No hosts provided."\
+                "At least one destination host is needed to install the RGW "\
+                "service")
+        self.ar_client.add_hosts_to_group(hosts, group)
+
+        # Limit playbook execution to certain hosts
+        limited = ",".join(hosts)
+
+        # Add the settings for this service
+        extravars = vars(spec.service_spec)
```

Comment:
I find this rather risky. We shouldn't pass data freely to any other service, specially when it comes from parsing a JSON file provided by the user (I know that the JSON file is validated against the RGW Spec class, but this is 

I'd deal with this issue by making:
-  `Orchestrator.StatelessServiceSpec` an abstract class (if not yet), and forcing to subclass it (e.g.: RGWServiceSpec) and implementing a method to only expose a subset of `public` and validated properties (e.g.: via `to_dict()` method).
- Using the Ansible adapter pattern mentioned above, which would help to retrieve all Ansible-related stuff (`limited`, `extravars` from `to_dict()`, etc.)